pkg/providers/v2/interface.go

```diff
@@ -41,6 +48,86 @@ var _ cloudprovider.Interface = (*cloud)(nil)
 type cloud struct {
 }
 
+type awsSDKProvider struct {
+	creds *credentials.Credentials
```

Comment:
Not sure we need a new type like `awsSDKProvider` yet, seems premature.  Can we have `creds` and the region field be in `cloud` instead?

Reply:
that sounds reasonable, will update!